core/src/tvu.rs

```diff
@@ -251,17 +251,11 @@ impl Tvu {
         // Before replay starts, set the callbacks in each of the banks in BankForks
         // Note after this callback is created, only the AccountsBackgroundService should be calling
         // AccountsDb::purge_slot() to clean up dropped banks.
-        let callback = bank_forks
-            .read()
+        let bank_drop_callback = SendDroppedBankCallback::new(pruned_banks_sender);
+        bank_forks
+            .write()
```

Comment:
Note that I'm grabbing a write lock on the bank-forks now.

I'm assuming since this is still during startup that there will not be any contention. I wanted to prevent any possibility of another thread holding a read lock and then getting a bank before the callbacks are set. This shouldn't be an issue though, since the ordering of the is-bank-drop-callback-enabled flag has been fixed. Wdyt? Seem ok?